### PR TITLE
Update comparison chart design and show it on the Product report

### PR DESCRIPTION
**Pull request by Aljullu**
Part of #695.

This PR does several things:
- Renders the charts with the comparison layout in the Product report.
- Updates the comparison layout to the new designs.
- Splits the CSS of the chart component in two different files (created a new one for the legend).

### Screenshots
image

### Detailed test instructions:
Given that the API endpoints are not ready yet, we must do a couple of tweaks in the code so we generate a long legend for the chart.

Go to `client/components/chart/legend.js` and replace [this line](https://github.com/woocommerce/wc-admin/compare/add/product-detail-report-chart?expand=1#diff-979f811d595d5bbcc243fc535a1ba3f0R83):
```ES6
const showTotalLabel = legendDirection === 'column' && data.length > 5 && itemsLabel;
```
with:
```ES6
const showTotalLabel = legendDirection === 'column' && data.length > 0 && itemsLabel;
```
and [this other line](https://github.com/woocommerce/wc-admin/compare/add/product-detail-report-chart?expand=1#diff-979f811d595d5bbcc243fc535a1ba3f0R102):
```ES6
{ data.map( row => ( 
```
with something like this:
```ES6
{ data.concat( data ).concat( data ).concat( data ).concat( data ).concat( data ).concat( data ).map( row => (
```
This way we will have many items in the legend and will be ready to test the PR.
- Go to the _Products_ report (`/wp-admin/admin.php?page=wc-admin#/analytics/products`).
- Verify the chart is displayed with the comparison layout (screenshot above):
  - The legend is on the left on wide screens and moves down to the bottom on mobile.
  - A shadow is displayed over the legend to make it clear it's scrollable and it disappears when you reach the end of the list.
  - A label is displayed at the bottom with the words _2 products_.
- Activate the Single Product filter (see screenshot).
image
  - Verify the label at the bottom of the legend changed to _2 variations_.

### To do
- The chart still displays all products data together instead of split by product/variation. I will leave #695 open to track that once the API allows querying the data that way.
- The chart height and the legend height don't match (as can be seen in the screenshot above). I will work on that in a follow-up PR, given that this one is already quite big.

### CODE REVIEW

**Comment by belcherj** [comment]
Getting this error on products screen:  `Error: <g> attribute transform: Trailing garbage, "translate(0,NaN)".`

**Comment by belcherj** [comment]
Should this be merged w/o apis being ready?

File: client/analytics/components/report-chart/index.js [hunk 1]
```diff
@@ -29,7 +29,15 @@ import ReportError from 'analytics/components/report-error';
 
 class ReportChart extends Component {
 	render() {
-		const { path, primaryData, secondaryData, selectedChart, query } = this.props;
+		const {
+			comparisonChart,
+			itemsLabel,
+			query,
```
Comment (belcherj) [inline client/analytics/components/report-chart/index.js]: not alpha sorted

File: client/analytics/report/products/index.js [hunk 2]
```diff
@@ -23,6 +24,11 @@ export default class ProductsReport extends Component {
 	render() {
 		const { path, query } = this.props;
 
+		const itemsLabel =
+			'single_product' === query.filter && !! query.products
+				? __( '%s variations', 'wc-admin' )
```
Comment (belcherj) [inline client/analytics/report/products/index.js]: What is the convention on multi-line ternaries?

File: client/components/chart/index.js [hunk 3]
```diff
@@ -211,12 +212,12 @@ class Chart extends Component {
 		chartHeight = width <= 783 ? 180 : chartHeight;
 		const legend = (
 			<Legend
-				className={ 'woocommerce-chart__legend' }
```
Comment (belcherj) [inline client/components/chart/index.js]: Was this deliberate?  Does this break other chart legends?

File: client/components/chart/legend.js [hunk 4]
```diff
@@ -14,7 +15,7 @@ import { formatCurrency } from '@woocommerce/currency';
 /**
  * Internal dependencies
  */
-import './style.scss';
+import './legend.scss';
```
Comment (belcherj) [inline client/components/chart/legend.js]: Why the change to sass file name?

File: client/components/chart/legend.js [hunk 5]
```diff
@@ -36,65 +37,108 @@ function getFormatedTotal( total, valueType ) {
...
+		this.updateListScroll();
+		window.addEventListener( 'resize', this.updateListScroll );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'resize', this.updateListScroll );
+	}
+
+	updateListScroll = () => {
+		const list = this.listRef.current;
```
Comment (belcherj) [inline client/components/chart/legend.js]: does react allow for request animation frame?

File: client/components/chart/index.js (hunk 3 above)
Comment (Aljullu) [reply client/components/chart/index.js]: Yes, that class is not used anywhere.

File: client/components/chart/legend.js (hunk 4 above)
Comment (Aljullu) [reply client/components/chart/legend.js]: `style.scss` was getting quite big so I decided to divide it in two files.

`style.scss` is still called from `charts.js` so it's not a problem removing it here.

**Comment by Aljullu** [comment]
> Getting this error on products screen: `Error: <g> attribute transform: Trailing garbage, "translate(0,NaN)".`

This seems to be related to the chart `<svg>` and it also occurs in master. I would say it's part of #582 and not related with this PR.

**Review by belcherj — APPROVED** [review]
LGTM

File: client/analytics/report/products/index.js (hunk 2 above)
Comment (Aljullu) [reply client/analytics/report/products/index.js]: Per https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#multi-line-statements I understand it's ok using them, right?

File: client/components/chart/legend.js (hunk 5 above)
Comment (Aljullu) [reply client/components/chart/legend.js]: I found something in the docs that might be useful in this particular case:
https://reactjs.org/docs/faq-functions.html#requestanimationframe-throttling

I spent some time playing with it but I don't have anything ready yet. I created an issue so we can take a look at it in the future: #822.

**Comment by Aljullu** [comment]
> Should this be merged w/o apis being ready?

We have been merging everything in #564 so I think it makes sense merging this one too.
